Clean up workload concept: job.md (#42677)
kubernetes/website: content/en/docs/concepts/workloads/controllers/job.md

consume.
{{< /note >}}

## Job patterns

The Job object can be used to support reliable parallel execution of Pods. The Job object is not

> designed to support closely-communicating parallel processes, as commonly found in scientific

Reviewer: NB: This is not true, but I'll deal with it in a separate PR.

Reviewer 2: Thanks

{{< feature-state for_k8s_version="v1.27" state="stable" >}}

> In most cases a parallel job will want the pods to run with constraints,

Reviewer: ```suggestion
In most cases, a parallel job will want the pods to run with constraints,
```

{{< feature-state for_k8s_version="v1.28" state="alpha" >}}

{{< note >}}
You can only set `podReplacementPolicy` on Jobs if you enable the `JobPodReplacementPolicy`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
{{< /note >}}

By default, the Job controller recreates Pods as soon they either fail or are terminating (have a deletion timestamp).

> This means that, at a given time, when some of the Pods are terminating, the number of running Pods for the Jobs

Reviewer: ```suggestion
This means that, at a given time, when some of the Pods are terminating, the number of running Pods for a Job
```

> can be greater than `parallelism` or greater than one Pod per index (if using Indexed Jobs).

Reviewer: ```suggestion
can be greater than `parallelism` or greater than one Pod per index (if you are using an Indexed Job).
```

You may choose to create replacement Pods only when the terminating Pod is fully terminal (has `status.phase: Failed`).
To do this, set the `.spec.podReplacementPolicy: Failed`.

> This will only recreate Pods once they are terminated.

Reviewer: let's remove this line. It's just repeating the previous line.
